File: AR-VR/devAyushDubey_ayushdubey_2024csit1122_2/Patterns/pattern15.cpp
#include <iostream>
using namespace std;
int main(){
    int n,i,j;
    cin >> n;
    int sp=n/2;
    int nm = 1;
    int val = 1;
    int cval;
    for(i=0;i<n;i++){
        cval = val;
        for(j=1;j<=sp;j++)
            cout<<"\t";
        for(j=1;j<=nm;j++){
            cout<<cval<<"\t";
            if(j<=nm/2)
                cval++;
            else
                cval--;
        }
        if(i<n/2){
            sp--;
            nm += 2;
            val++;
        }
        else{
            sp++;
            nm -= 2;
            val--;
        }
        cout<<endl;
    }
    //write your code here
    
}

File: IOT/shakap12_ShashwatKapoor_1923ec1177_3/Patterns/Pattern15.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        // write ur code here
        int n=scn.nextInt();
        int spaces=n/2;
        int row=1;
        int val=1;
        for(int i=0;i<n;i++){
            for(int j=0;j<spaces;j++){
                System.out.print("\t");
            }
            int c=val;
            for(int k=0;k<row;k++){
                System.out.print(c+"\t");
                if(k<row/2)
                c++;
                else if(k>=row/2)
                c--;
            }
            
            if(i<n/2){
                spaces--;
                row+=2;
                val++;
            }
            else{
                spaces++;
                row-=2;
                val--;
            }
            System.out.println();
        }

    }
}